### PR TITLE
refactor(templates): use Vue CLI for 2.x.x

### PR DESCRIPTION
**Pull request by samouss**
This PR updates the template for Vue InstantSearch 2.x.x with Vue CLI. 

I've update the `.eslintrc.js` to disable two rules that conflict each other (until we fix the Algolia one):

- extends from `prettier/vue`
- use kebab-case for `vue/component-name-in-template-casing`

### CODE REVIEW

File: src/templates/Vue InstantSearch/package.json [hunk 1]
```diff
@@ -3,36 +3,34 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "start": "cross-env NODE_ENV=development webpack-dev-server --port 3000 --hot",
-    "build": "cross-env NODE_ENV=production webpack --progress --hide-modules",
-    "lint": "eslint --ext .js,.vue .",
+    "start": "vue-cli-service serve",
```
Comment (francoischalifour) [inline src/templates/Vue InstantSearch/package.json]: The port for all generated apps is `3000`.

**Comment by Haroenv** [comment]
manifest.json is removed because Vue CLI no longer generates it I think

**Comment by samouss** [comment]
> The app is terribly lagging on my machine. Are you experiencing the same thing?

Nope, I've also upload it to [CodeSandbox](https://codesandbox.io/s/43202po689) and I don't have the issue.

> Why did you remove `manifest.json`?

Yep @Haroenv is correct, it's only generated with the PWA plugin enabled.

> https://cli.vuejs.org/guide/html-and-static-assets.html#the-index-file

**Comment by samouss** [comment]
Prettier have a conflict with the Vue ESLint plugin at the moment we'll see how to fix it.

File: src/templates/Vue InstantSearch/package.json (hunk 1 above)
Comment (francoischalifour) [inline src/templates/Vue InstantSearch/package.json]: Let's change this for consistency with other templates and to align with the README instructions.

```suggestion
    "start": "vue-cli-service serve --port 3000",
```

**Comment by francoischalifour** [comment]
We can update the Vue InstantSearch v1 template is another PR.

File: src/templates/Vue InstantSearch/package.json (hunk 1 above)
Comment (samouss) [reply src/templates/Vue InstantSearch/package.json]: I've changed it locally, but I discover a conflict with Prettier on my local setup. The service lint correctly but not the editor. Do you any issue?

Comment (francoischalifour) [reply src/templates/Vue InstantSearch/package.json]: I'm not sure I understand. What are the steps to reproduce?

Comment (samouss) [reply src/templates/Vue InstantSearch/package.json]: Whenever I save `App.vue` the lint is broken (I have Prettier that runs on save). If I run the lint service of the template everything is correct though.

**Comment by francoischalifour** [comment]
Good call on 1e6473312cb278681999801766fba348f7bb897e @Haroenv 👍

File: src/templates/Vue InstantSearch/src/App.vue [hunk 2]
```diff
@@ -27,7 +29,9 @@
 
           {{/if}}
           <div class="search-panel__results">
-            <ais-search-box placeholder="{{searchPlaceholder}}" />
+            <div class="searchbox">
```
Comment (francoischalifour) [inline src/templates/Vue InstantSearch/src/App.vue]: Nicely spotted!

**Pull request merged**